opencog/atomutils/Unify.cc

```diff
@@ -348,14 +355,126 @@ bool Unify::is_bound_to_ancestor(const Variables& variables,
 
 Handle Unify::substitute(BindLinkPtr bl, const TypedSubstitution& ts)
 {
-	// Perform alpha-conversion, this will work over values that are
-	// non variables as well
-	//
 	// TODO: make sure that ts.second contains the declaration of all
 	// variables
-	Handle h = bl->alpha_conversion(strip_context(ts.first), ts.second);
+	return substitute(bl, strip_context(ts.first), ts.second);
+}
+
+Handle Unify::substitute(BindLinkPtr bl, const HandleMap& var2val,
+                         Handle vardecl)
```

Comment:
I thought of `beta_reduce()` but again was afraid of conflating with the existing methods of that name. I though of `partial_substitute()` as well. Maybe `partial_beta_reduce()` could be appropriate.